runtime/runtime/src/actions.rs

```diff
@@ -310,64 +334,78 @@ pub(crate) fn action_delete_account(
     Ok(())
 }
 
-// TODO(#1461): Add Safe Math
 pub(crate) fn action_delete_key(
     fee_config: &RuntimeFeesConfig,
     state_update: &mut TrieUpdate,
-    account: &mut Option<Account>,
+    account: &mut Account,
     result: &mut ActionResult,
     account_id: &AccountId,
     delete_key: &DeleteKeyAction,
 ) -> Result<(), StorageError> {
-    let account = account.as_mut().unwrap();
-    let access_key = get_access_key(state_update, account_id, &delete_key.public_key)?;
-    if access_key.is_none() {
-        result.result = Err(ActionError::DeleteKeyDoesNotExist(account_id.clone()));
-        return Ok(());
-    }
+    let access_key = match get_access_key(state_update, account_id, &delete_key.public_key)? {
+        Some(access_key) => access_key,
+        None => {
+            result.result = Err(ActionError::DeleteKeyDoesNotExist(account_id.clone()));
+            return Ok(());
+        }
+    };
     // Remove access key
     state_update.remove(&key_for_access_key(account_id, &delete_key.public_key));
     let storage_usage_config = &fee_config.storage_usage_config;
-    account.storage_usage -= (delete_key.public_key.try_to_vec().ok().unwrap_or_default().len()
-        as u64)
-        * storage_usage_config.key_cost_per_byte;
-    account.storage_usage -= (access_key.unwrap().try_to_vec().ok().unwrap_or_default().len()
-        as u64)
-        * storage_usage_config.value_cost_per_byte;
-    account.storage_usage -= storage_usage_config.data_record_cost;
+    account.storage_usage = account
+        .storage_usage
+        .checked_sub(
+            (delete_key.public_key.try_to_vec().unwrap().len() as u64)
+                * storage_usage_config.key_cost_per_byte
```

Comment:
Should we use safe math here and below?

Reply:
This is actually unlikely to overflow because the public key can't be large enough to overflow.